plugins/example-script-service/src/main/java/org/elasticsearch/plugin/example/ExampleScriptService.java

```diff
@@ -0,0 +1,110 @@
+import org.elasticsearch.script.CompiledScript;
+import org.elasticsearch.script.ExecutableScript;
+import org.elasticsearch.script.ScriptEngineService;
+import org.elasticsearch.script.SearchScript;
+import org.elasticsearch.search.lookup.SearchLookup;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Example of a simple script service that adds several pre-defined scripts
+ */
+public class ExampleScriptService implements ScriptEngineService {
+
+    public static final String NAME = "example";
+
+    public static final String NAME_EXT = "xmpl";
+
+    public ExampleScriptService(Settings settings) {
+        // TODO: Show how to access settings
+    }
+
+    /**
+     * This name is used to identify the script engine in elasticsearch
+     */
+    @Override
+    public String getType() {
+        return NAME;
+    }
+
+    /**
+     * We don't add a scripting language in this example storing these scripts on file system doesn't make
+     * much sense. But we need to define a unique extension, so it wouldn't clash with other script engines.
+     */
+    @Override
+    public String getExtension() {
+        return NAME_EXT;
+    }
+
+    /**
+     * This method is called every time elasticsearch encounters a new script. The returned object is cached and then
+     * given back to use for execution either using {@link #executable(CompiledScript, Map)} or
+     * {@link #search(CompiledScript, SearchLookup, Map)} methods depending on context.
+     *
+     * @param scriptName - the name of the file where script was stored, this parameter is ignored in most cases
+     * @param scriptSource - the actual script
+     * @param params - the script compilation parameters, in most cases this set of parameters is empty. This is NOT the user-specified
+     *               list of parameters which will be provided in executable and search method.
+     * @return the compiled version of the script suitable for caching
+     */
+    @Override
+    public Object compile(String scriptName, String scriptSource, Map<String, String> params) {
+        switch (scriptSource) {
+            case "is_holiday":
+                return new IsHolidayScriptFactory();
+            default:
+                throw new IllegalArgumentException("Script [" + scriptSource + "] not found");
+        }
+    }
+
+    @Override
+    public ExecutableScript executable(CompiledScript compiledScript, @Nullable Map<String, Object> vars) {
+        ExampleScriptFactory scriptFactory = (ExampleScriptFactory) compiledScript.compiled();
+        return scriptFactory.executable(vars);
+    }
+
+    @Override
+    public SearchScript search(CompiledScript compiledScript, SearchLookup lookup, @Nullable Map<String, Object> vars) {
+        ExampleScriptFactory scriptFactory = (ExampleScriptFactory) compiledScript.compiled();
+        return scriptFactory.search(lookup, vars);
+    }
+
+    @Override
+    public void close() throws IOException {
+
```

Comment:
Add a comment here explaining script engines are Closeable and resources can be released here?